On amd64 linux /usr/lib does not match default 64-bit ABI.

Let's rely on default compiler and linker paths instead.
--- a/src/compiler/crystal/codegen/link.cr
+++ b/src/compiler/crystal/codegen/link.cr
@@ -105,7 +105,7 @@ module Crystal
     end
 
     private def lib_flags_posix
-      library_path = ["/usr/lib", "/usr/local/lib"]
+      library_path = [] of String
       has_pkg_config = nil
 
       String.build do |flags|
